pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2244,9 +2244,18 @@ private CompletableFuture<Integer> unloadServiceUnit(NamespaceBundle serviceUnit
                 closeFutures.add(topicFuture
                         .thenCompose(t -> t.isPresent() ? t.get().close(
                                 disconnectClients, closeWithoutWaitingClientDisconnect)
-                                : CompletableFuture.completedFuture(null)));
+                                : CompletableFuture.completedFuture(null))
+                        .exceptionally(e -> {
+                            if (e.getCause() instanceof BrokerServiceException.ServiceUnitNotReadyException
+                                    && e.getMessage().contains("Please redo the lookup")) {
+                                log.warn("[{}] Topic ownership check failed. Skipping it", topicName);
+                                return null;
```

Comment:
Since we're adding a null element to `closeFutures`, would this cause an exception further down in `FutureUtil.waitForAll`? The [documentation](https://docs.oracle.com/en%2Fjava%2Fjavase%2F11%2Fdocs%2Fapi%2F%2F/java.base/java/util/concurrent/CompletableFuture.html#allOf(java.util.concurrent.CompletableFuture...)) suggests so.

Reply:
This `return null` does not return a null element to `closeFutures`. I think it will add CompletableFuture(null).

Reply:
You're right, thanks for clarifying!